File: android/dragnslay/src/net/kibotu/sandbox/unity/android/dragnslay/utility/DisplayHelper.java
package net.kibotu.sandbox.unity.android.dragnslay.utility;

import android.graphics.Point;
import net.kibotu.sandbox.unity.android.dragnslay.UnityPlayerNativeActivity;

public class DisplayHelper {

    private DisplayHelper() {
    }

    public static int getDisplayWidth() {
        return getSize().x;
    }

    public static int getDisplayHeight() {
        return getSize().y;
    }

    public static Point getSize() {
        Point ret = new Point();
        UnityPlayerNativeActivity.context.getWindowManager().getDefaultDisplay().getSize(ret);
        return ret;
    }
}

File: unity/Assets/Src/net/kibotu/sandbox/unity/dragnslay/utility/DisplayHelper.cs
namespace Assets.Src.net.kibotu.sandbox.unity.dragnslay.utility
{
    class DisplayHelper
    {
        // static
        private DisplayHelper()
        {
        }

        public static int GetDisplayWidth()
        {
            #if UNITY_EDITOR
                
                return 480;
               
            #elif UNITY_ANDROID
            
                var displayHelper = new AndroidJavaClass("net.kibotu.sandbox.android.DisplayHelperClass");
                var width = displayHelper.CallStatic<int>("getDisplayWidth");
                var height = displayHelper.CallStatic<int>("getDisplaHeight");
                Debug.Log("Display [w=" + width + "|h=" + height + "]");
            
            #endif
        }
    }
}
